app/models/branch.rb

```diff
@@ -0,0 +1,10 @@
+class Branch < ApplicationRecord
+
+  has_many :reservations, dependent: :restrict_with_error
+  has_many :tables, dependent: :destroy
+  has_many :products, -> { includes(:product) }, class_name: :BranchProduct, dependent: :destroy
```

Comment:
what is includes doing here
avoid adding includes like this
add it only if needed

Reply:
Sir, previously `:products` association was giving us the branch products and we were having another association of `:product` on `branch_products`. So, I was using includes to eager load products on the `branch_products`.

Now fixed this all and associated these with has_many through, so having a direct association with products now.

Reply:
let's discuss